internal/bgp/bgpfrr/config.go

```diff
@@ -0,0 +1,127 @@
+package bgpfrr
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"text/template"
+)
+
+// TODO configFileName will need to be set to the filename of the FRR configuration
+// file in the container.
+var configFileName = "frr.conf"
+
+const configTemplate = `
+log stdout {{.Loglevel}}
+hostname {{.Hostname}}
+
+{{range .Routers -}}
+router bgp {{.Asn}}
+  no bgp default ipv4-unicast
+  bgp router-id {{.RouterId}}
+{{range .Neighbors }}
+  neighbor {{.Addr}} remote-as {{.Asn}}
+{{- end }}
+{{range $n := .Neighbors -}}
+{{range .Advertisements }}
+  address-family {{.Version}} unicast
+    neighbor {{$n.Addr}} activate
+    network {{.Prefix}}
+  exit-address-family
+{{- end}}
+{{end}}
+{{end}}
+`
+
+type frrConfig struct {
+	Loglevel string
+	Hostname string
+	Routers  map[string]*routerConfig
+}
+
+type routerConfig struct {
+	Asn       uint32
+	RouterId  string
+	Neighbors map[string]*neighborConfig
+}
+
+type neighborConfig struct {
+	Asn            uint32
+	Addr           string
+	Advertisements map[string]*advertisementConfig
+}
+
+type advertisementConfig struct {
+	Version string
+	Prefix  string
+}
+
+// routerName() defines the format of the key of the "Routers" map in the
+// frrConfig struct.
+func routerName(srcAddr string, myAsn uint32) string {
+	return fmt.Sprintf("%d@%s", myAsn, srcAddr)
+}
+
+// neighborName() defines the format of key of the 'Neighbors' map in the
+// routerConfig struct.
+func neighborName(peerAddr string, peerAsn uint32) string {
+	return fmt.Sprintf("%d@%s", peerAsn, peerAddr)
+}
```

Comment:
peerAddr should be unique enough I don't think its normal/possible to have same neighbor defined in the same speaker if its for different remote-as ?

Reply:
I think it is possible (at least in FRR). See https://docs.frrouting.org/en/latest/bgp.html#clicmd-router-bgp-ASN-vrf-VRFNAME

Reply:
yeah under different VRF sure but we don't have VRF support at all , correct ?

Reply:
There are a few comments here which are basically discussing requirements. I am unsure what the requirements are and, IMHO, I think they should be defined by our BGP tests. Can I propose that I resolve these comments in this peer and I will raise them as comments in the PR defining BGP test cases? @msherif1234 @fedepaol what do you think?

Reply:
Fine by me. The api allows it. Let's add a test covering this scenario so we both validate our assumption and we match the current behaviour.

Reply:
@msherif1234 ?